src/pages/Signup/Signup.js

```diff
@@ -1,7 +1,185 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 import './Signup.scss';
+
 const Signup = () => {
-  return <div>Signup</div>;
+  const [userInfo, setUserInfo] = useState({
+    name: '',
+    account: '',
+    email: '',
+    phone_number: '',
+    password: '',
+  });
+
+  const navigate = useNavigate();
+
+  const { name, account, email, phone_number, password } = userInfo;
+
+  const saveUserInfo = e => {
+    const { name, value } = e.target;
+    setUserInfo({ ...userInfo, [name]: value });
+  };
+
+  const accountRegExp = /^[a-z0-9+_.]{4,}/;
+  const emailRegExp = /^([a-z0-9_.-]+)@([\da-z.-]+)\.([a-z.]{2,6})$/;
+  const phoneNumberRegExp = /[0-9]{10,11}/;
+  const passwordRegExp = /^[a-z0-9_-]{8,}$/;
+
+  const isAccountValid = accountRegExp.test(account);
+  const isEmailValid = emailRegExp.test(email);
+  const isPhoneNumberValid = phoneNumberRegExp.test(phone_number);
+  const isPasswordValid = passwordRegExp.test(password);
+
+  const redWarningAccount =
+    account.length === 0 || isAccountValid
+      ? 'redMessageInactive'
+      : 'redMessageActive';
+
+  const redWarningEmail =
+    email.length === 0 || isEmailValid
+      ? 'redMessageInactive'
+      : 'redMessageActive';
+
+  const redWarningPhoneNumber =
+    phone_number.length === 0 || isPhoneNumberValid
+      ? 'redMessageInactive'
+      : 'redMessageActive';
+
+  const redWarningPassword =
+    password.length === 0 || isPasswordValid
+      ? 'redMessageInactive'
+      : 'redMessageActive';
+
+  const postUserData = () => {
+    fetch('http://10.58.3.43:8000/users/signup', {
+      method: 'POST',
+      body: JSON.stringify({
+        name: name,
+        account: account,
+        email: email,
+        phone_number: phone_number,
+        password: password,
+      }),
```

Comment:
```suggestion
      body: JSON.stringify({
        name,
        account,
        email,
        phone_number,
        password,
      }),
```

객체의 key, value가 같은경우에는
`단축속성명` (shorthand property name) 으로 줄여서 쓸 수 있습니다.

Reply:
✅